proxy/proxy.go

```diff
@@ -12,27 +12,33 @@ import (
 
 // MakeHTTPClientWithDisableKeepAlives makes a HTTP client with good defaults for timeouts.
 func MakeHTTPClientWithDisableKeepAlives(timeout *time.Duration, tlsInsecure bool, disableKeepAlives *bool) http.Client {
-	if timeout != nil {
-		tr := &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-			DialContext: (&net.Dialer{
+	client := http.Client{}
+
+	if timeout != nil || tlsInsecure || disableKeepAlives != nil {
+		tr := &http.Transport{Proxy: http.ProxyFromEnvironment}
+
+		if timeout != nil {
+			client.Timeout = *timeout
+			tr.DialContext = (&net.Dialer{
 				Timeout: *timeout,
-			}).DialContext,
+			}).DialContext
+
+			tr.IdleConnTimeout = 120 * time.Millisecond
+			tr.ExpectContinueTimeout = 1500 * time.Millisecond
+		}
 
-			IdleConnTimeout:       120 * time.Millisecond,
-			ExpectContinueTimeout: 1500 * time.Millisecond,
-			TLSClientConfig:       &tls.Config{InsecureSkipVerify: tlsInsecure},
+		if tlsInsecure {
```

Comment:
Is it harmful to set `tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: false`?  If not, do we need the `if` here?

Reply:
@rgee0 i'm not sure of the side effects:

> // TLSClientConfig specifies the TLS configuration to use with
	// tls.Client.
	// If nil, the default configuration is used.
	// If non-nil, HTTP/2 support may not be enabled by default.
	TLSClientConfig *tls.Config

From above:
>  If nil, the default configuration is used.

I can't find the default configuration for `TLSConfig` but there are a lot of fields on that struct so could potentially change behaviour

Reply:
I was thinking that `tlsInsecure` defaults to `false` and is set to `true` via the flag, so it should always have a non-nil value.  To that end if `tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: false}` is not harmful then testing the value is true before setting might be unnecessary.